### PR TITLE
Fix issues found with phpstan 0.10.1

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- An error message about subclasses has been fixed
```

### CODE REVIEW

File: src/Admin/AdminInterface.php [hunk 1]
```diff
@@ -335,7 +335,7 @@ public function setUniqid($uniqId);
     /**
      * Returns the uniqid.
      *
-     * @return int
+     * @return string
      */
     public function getUniqid();
```
Comment (OskarStark) [inline src/Admin/AdminInterface.php]: should be `getUniqId` and `setUniqId` IMO

Comment (greg0ire) [reply src/Admin/AdminInterface.php]: Maybe the goal was consistency with https://secure.php.net/manual/en/function.uniqid.php ?

Comment (OskarStark) [reply src/Admin/AdminInterface.php]: yeah maybe, no need to care about to long ;-)
Lets keep it

File: src/Admin/AbstractAdmin.php [hunk 2]
```diff
@@ -284,7 +284,7 @@
     /**
      * The current request object.
      *
-     * @var \Symfony\Component\HttpFoundation\Request
+     * @var ?\Symfony\Component\HttpFoundation\Request
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: This is valid PHPDoc?

Comment (greg0ire) [reply src/Admin/AbstractAdmin.php]: I tried using phpdocumentor on this and it does weird things like `\?\Datetime`, I'm going to use `|null` instead

File: src/Admin/AbstractAdmin.php [hunk 3]
```diff
@@ -284,7 +284,7 @@
     /**
      * The current request object.
      *
-     * @var \Symfony\Component\HttpFoundation\Request
+     * @var \Symfony\Component\HttpFoundation\Request|null
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: Please don't use FCQN here.

File: src/Admin/AbstractAdmin.php [hunk 4]
```diff
@@ -335,14 +335,14 @@
     /**
      * The datagrid instance.
      *
-     * @var \Sonata\AdminBundle\Datagrid\DatagridInterface
+     * @var \Sonata\AdminBundle\Datagrid\DatagridInterface|null
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: And here...

File: src/Admin/FieldDescriptionInterface.php [hunk 5]
```diff
@@ -50,7 +50,7 @@ public function getName();
      * @param string     $name
      * @param mixed|null $default
      *
-     * @return array|null the value represented by the provided name
+     * @return mixed the value represented by the provided name
```
Comment (core23) [inline src/Admin/FieldDescriptionInterface.php]: According to PHPDoc, does `mixed` include `null` values?

File: src/Command/GenerateObjectAclCommand.php [hunk 6]
```diff
@@ -97,6 +98,7 @@ public function execute(InputInterface $input, OutputInterface $output)
                 continue;
             }
 
+            assert($admin instanceof AdminInterface);
```
Comment (core23) [inline src/Command/GenerateObjectAclCommand.php]: Should we use `assert`s in general? Symfony is not using asserts in any of its code.

File: src/Controller/CRUDController.php [hunk 7]
```diff
@@ -1525,7 +1529,7 @@ private function checkParentChildAssociation(Request $request, $object)
     /**
      * Sets the admin form theme to form view. Used for compatibility between Symfony versions.
      *
-     * @param string $theme
+     * @param array $theme
      */
     private function setFormTheme(FormView $formView, $theme)
```
Comment (core23) [inline src/Controller/CRUDController.php]: You could add `array` infront of the parameter, because this method is private

File: src/Admin/FieldDescriptionInterface.php (hunk 5 above)
Comment (greg0ire) [reply src/Admin/FieldDescriptionInterface.php]: Yes, see https://docs.phpdoc.org/guides/types.html

File: src/Admin/AbstractAdmin.php (hunk 3 above)
Comment (greg0ire) [reply src/Admin/AbstractAdmin.php]: Not my fault but ok

File: src/Command/GenerateObjectAclCommand.php (hunk 6 above)
Comment (greg0ire) [reply src/Command/GenerateObjectAclCommand.php]: I've tried it at work, and I think it's slightly better than comments because:

- it might get checked if a dev has assertions enabled
- it can be used to have intersection types without resorting to not-fully-supported-everywhere phpdoc like `/** @var A&B */`
More on this here: https://github.com/doctrine/coding-standard/pull/47

File: src/Admin/AbstractAdmin.php (hunk 3 above)
Comment (core23) [reply src/Admin/AbstractAdmin.php]: I know, but you touched this line, so you could do some cleanup as well :)

File: src/Admin/FieldDescriptionInterface.php (hunk 5 above)
Comment (core23) [reply src/Admin/FieldDescriptionInterface.php]:
```
mixed
    A value with this type can be literally anything; the author of the documentation is unable to predict which type it will be.
```

IMHO *anything* doesn't mean nothing (null).

Comment (greg0ire) [reply src/Admin/FieldDescriptionInterface.php]: I think nothing is better representing by void, also, if you look at the list of types, null is in it, while void isn't.

> The value contained, or returned, is literally null. This type is not to be confused with void, which is the total absence of a variable or value (usually used with the @return tag).

File: src/Command/GenerateObjectAclCommand.php (hunk 6 above)
Comment (greg0ire) [reply src/Command/GenerateObjectAclCommand.php]: Also, the Travis build fails because this assertion is incorrect, which shows why assertions are better ;)

File: src/Admin/AbstractAdmin.php (hunk 3 above)
Comment (greg0ire) [reply src/Admin/AbstractAdmin.php]: I could, and did, wait for the push :P

File: src/Command/GenerateObjectAclCommand.php (hunk 6 above)
Comment (greg0ire) [reply src/Command/GenerateObjectAclCommand.php]: Turns out this one is very very hard to fix, I'll just remove it for now, but there's something quite wrong in there: the container throws on missing service, while our tests mock it so that it returns null…

Comment (greg0ire) [reply src/Command/GenerateObjectAclCommand.php]: Ah it's not this one, it's the one in Pool::getInstance, sorry.

File: src/Admin/AbstractAdmin.php [hunk 8]
```diff
@@ -2733,7 +2740,7 @@ public function getDashboardActions()
     }
 
     /**
-     * @param FormMapper $form
+     * @param FormMapper $object
```
Comment (core23) [inline src/Admin/AbstractAdmin.php]: Sure this is a `FormMapper` instance?

File: src/Controller/CRUDController.php [hunk 9]
```diff
@@ -330,7 +331,7 @@ public function editAction($id = null)
         $this->admin->setSubject($existingObject);
         $objectId = $this->admin->getNormalizedIdentifier($existingObject);
 
-        /** @var $form Form */
+        /** @var Form */
```
Comment (core23) [inline src/Controller/CRUDController.php]: Do we need this PHPDoc at all?

File: src/Controller/CRUDController.php [hunk 10]
```diff
@@ -585,7 +586,7 @@ public function createAction()
 
         $this->admin->setSubject($newObject);
 
-        /** @var $form \Symfony\Component\Form\Form */
+        /** @var \Symfony\Component\Form\Form */
```
Comment (core23) [inline src/Controller/CRUDController.php]: Do we need this PHPDoc at all?

File: src/Controller/CRUDController.php [hunk 11]
```diff
@@ -1530,10 +1532,8 @@ private function checkParentChildAssociation(Request $request, $object)
 
     /**
      * Sets the admin form theme to form view. Used for compatibility between Symfony versions.
-     *
-     * @param string $theme
      */
-    private function setFormTheme(FormView $formView, $theme)
+    private function setFormTheme(FormView $formView, array $theme = null)
```
Comment (kunicmarko20) [inline src/Controller/CRUDController.php]: Isn’t this a BC break?

**Comment by OskarStark** [comment]
Thank you @greg0ire 👍
